[Feat/application_view] application_view pagenation 로직 추가 및 테스트 코드 추가

Kang9805 opened the pull request:
## ✅ PR 요약
- 작업 요약: application view와 test code 추가

## 📄 상세 내용
- [ ]  admin_application_view  schema에 페이지네이션 미표시 문제를 해결
- [ ] 검색(search), 정렬(sort), 상태(status) 필터를 명세서에 맞게 통합
- [ ] application_view 기존 TODO로 표시 되어있던 페이지네이션 해결
- [ ] test application serializers, test admin application serializers 유닛 테스트 코드 작성

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Kang9805 committed:
✨ feat : 내 지원 내역 목록에 CursorPagination 적용 및 구조 개선

- MyApplicationListView에 CursorPagination 도입
- select_related("recruitment") 추가하여 조회 성능 최적화

Kang9805 committed:
✅ test :  ApplicantApplication - Pagination 테스트 추가

- MyApplicationListView cursor 기반 페이지네이션 테스트 추가

Kang9805 committed:
♻️ refactor : applicant views와 admin application views pagination 로직 정비

- AdminApplicationListView 페이지네이션 로직 수정
- search 파라미터 통합 검색 처리 반영
- schema 파라미터 및 response 구조 정리

Kang9805 committed:
✅ test : application/admin application serializer 테스트 코드 작성

- 실제 Serializer 동작과 동일하게 mock 객체 업데이트

SWWS97 commented on apps/application/views/admin_application_views.py:
```diff
@@ -39,30 +39,30 @@ class AdminApplicationListView(APIView):
 
-        # 검색 (공고 제목, 지원자 닉네임, 지원자 이메일)
+        # 통합 검색 (공고 제목, 지원자 닉네임, 지원자 이메일)
         search_keyword = request.GET.get("search")
```
위에선 query_params로 조회하는데 밑에선 search_keyword = request.GET.get("search") 이렇게 작성한 이유가 뭐에요?

Kang9805 replied:
통일 시키도록 하겠습니다

Kang9805 committed:
♻️ refactor : 지원 삭제/취소/생성 로직 수정 및 id 기반 조회로 통일

- applicant 상세조회/삭제 엔드포인트를 id 기반으로 일관성 있게 변경
- 명세서와 테스트 요구사항에 맞게 오류 메시지 통일

Kang9805 committed:
♻️ refactor : 관리자용 지원서 상세조회 id 기반으로 변경

- AdminApplicationDetailView에서 uuid → id 조회로 수정
- 존재하지 않는 지원서 404 메시지 명세서와 일치하도록 수정

Kang9805 committed:
✨ feat : 지원서 삭제 DELETE 엔드포인트 생성

- ApplicationCreateView의 recruitment_uuid URL 패턴 유지
- 명세서에 맞게 전체 URL 구조 통일
- 지원서 삭제 엔드포인트 생성

Kang9805 committed:
✅ test : 바뀐 view에 맞게 테스트 코드 수정

- applicant 지원서 API 전체 테스트 리팩터링
- 상세조회 및 목록 조회 테스트 수정
